CoreScriptsRoot/CoreScripts/HealthScript.lua

```diff
@@ -262,55 +260,71 @@ function newPlayerCharacter()
 end
 
 function startGui()
-	characterAddedConnection = game:GetService("Players").LocalPlayer.CharacterAdded:connect(newPlayerCharacter)
+	characterAddedConnection = Players.LocalPlayer.CharacterAdded:connect(newPlayerCharacter)
 
-	local character = game:GetService("Players").LocalPlayer.Character
+	local character = Players.LocalPlayer.Character
 	if not character then
-		return
+		return nil
 	end
 
 	currentHumanoid = character:WaitForChild("Humanoid")
 	if not currentHumanoid then
-		return
+		return nil
 	end
 
-	if not game:GetService("StarterGui"):GetCoreGuiEnabled(Enum.CoreGuiType.Health) then
-		return
+	if not StarterGui:GetCoreGuiEnabled(Enum.CoreGuiType.Health) then
+		if useDefault == false then
+			return nil
+		end		
 	end
-
+	
 	healthChangedConnection = currentHumanoid.HealthChanged:connect(UpdateGui)
 	humanoidDiedConnection = currentHumanoid.Died:connect(humanoidDied)
 	UpdateGui(currentHumanoid.Health)
-		
 	CreateGui()
 end
 
-
-
 ---------------------------------------------------------------------
 -- Start Script
+---------------------------------------------------------------------
 
-HealthGui = Instance.new("Frame")
-HealthGui.Name = "HealthGui"
-HealthGui.BackgroundTransparency = 1
-HealthGui.Size = UDim2.new(1,0,1,0)
+function begin()
+	if StarterGui:GetCoreGuiEnabled(Enum.CoreGuiType.Health) == true or StarterGui:GetCoreGuiEnabled(Enum.CoreGuiType.Health) == true then
+		if useDefault == false then
+			guiEnabled = true
+			HealthGui = Instance.new("ScreenGui")
+			HealthGui.Name = "HealthGui"
+			HealthGui.Parent = Player.PlayerGui
+			startGui()
+		else
+			guiEnabled = true
+			HealthGui = Instance.new("ScreenGui")
+			HealthGui.Name = "HealthGui"
+			HealthGui.Parent = CoreGui.RobloxGui
+			startGui()
+		end
+	elseif StarterGui:GetCoreGuiEnabled(Enum.CoreGuiType.Health) == false or StarterGui:GetCoreGuiEnabled(Enum.CoreGuiType.Health) == false then
+		if useDefault == false then
+			guiEnabled = true
+			HealthGui = Instance.new("ScreenGui")
+			HealthGui.Name = "HealthGui"
+			HealthGui.Parent = Player.PlayerGui
+			startGui()
+		end
+	end
+end
 
-game:GetService("StarterGui").CoreGuiChangedSignal:connect(function(coreGuiType,enabled)
+pcall(function()StarterGui.CoreGuiChangedSignal:connect(function(coreGuiType,enabled)
 	if coreGuiType == Enum.CoreGuiType.Health or coreGuiType == Enum.CoreGuiType.All then
-		if guiEnabled and not enabled then
-			if HealthGui then
-				HealthGui.Parent = nil
-			end
-			disconnectPlayerConnections()
-		elseif not guiEnabled and enabled then
-			startGui()
+		if game:GetService("StarterGui"):GetCoreGuiEnabled(Enum.CoreGuiType.Health) == true then
+			begin()
+		elseif game:GetService("StarterGui"):GetCoreGuiEnabled(Enum.CoreGuiType.Health) == false then
+			begin()
 		end
-		
-		guiEnabled = enabled
 	end
 end)
+end)
 
-if game:GetService("StarterGui"):GetCoreGuiEnabled(Enum.CoreGuiType.Health) then
-	guiEnabled = true
-	startGui()
-end
+begin()
+
+return true
```

Comment:
I thought all core scripts where modules. That had localscripts require() them. Am I wrong?